Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo

```diff
@@ -33,12 +33,14 @@ algorithm
     //Limited relative reference signal
     uRef:=min(sqrt(u[1]^2 + u[2]^2)/(2/3*uMax), cos(pi/6));
     //Determine angle of reference signal within (-pi, +pi]
-    phiRef:=if noEvent(uRef < small) then 0 else atan2(u[2], u[1]);
+    phiRef:=if uRef < small then 0 else atan2(u[2], u[1]);
     //Shift angle to [0, 2*pi)
-    phiPos:=max(phiRef + (if phiRef < -eps then 2*pi else 0), 0);
+    // Note that phiRef is likely to be very close to zero, so the extra logic ensures that
+    // we do not get values at 2*pi since that will look odd.
+    phiPos:=max(0, phiRef + (if phiRef < -2*pi*Modelica.Constants.eps then 2*pi else 0));
```

Comment:
I can see the logic of choosing the factor `2 ^ pi` in front of `eps`, but it is probably not obvious to someone who isn't familiar with the peculiar definition of `eps` in the MSL.  Might be worth explaining what's going on in a comment…

Reply:
Could Modelica.Math.wrapAngle be utilized here?

Reply:
nope, tested: results in "Index out of bounds".